pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -74,16 +77,104 @@ type cloudWatchExecutor struct {
 
 func (e *cloudWatchExecutor) newSession(region string) (*session.Session, error) {
 	dsInfo := e.getDSInfo(region)
-	creds, err := getCredentials(dsInfo)
+
+	bldr := strings.Builder{}
+	for i, s := range []string{dsInfo.AuthType.String(), dsInfo.AccessKey, dsInfo.Profile, dsInfo.AssumeRoleARN} {
+		if i != 0 {
+			bldr.WriteString(":")
+		}
+		bldr.WriteString(strings.ReplaceAll(s, ":", `\:`))
+	}
+	cacheKey := bldr.String()
+	plog.Debug("Using cache key", "key", cacheKey)
```

Comment:
This message is a bit misleading: the key will be only used if it will be found in the cache.
Anyway, given the whole discussion about logging maybe we should consider not keeping it.